delly.wdl

```diff
@@ -1,16 +1,37 @@
 version 1.0
 
+struct GenomeResources {
+    String rundelly_module
+    String rundelly_fasta
+    String rundelly_exclude_list
+}
+
 workflow delly {
+
 input {
   # If we are in somatic mode, normal file follows tumor file in the input array
   File inputTumor
   File? inputNormal
   Boolean markdup = true
-  String outputFileNamePrefix = ""
+  String outputFileNamePrefix
+  String reference
+}
+
+Map[String,GenomeResources] resources = {
+  "hg19": {
+    "rundelly_module": "delly/0.9.1 bcftools/1.9 tabix/0.2.6 hg19/p13 hg19-delly/1.0",
+    "rundelly_fasta": "$HG19_ROOT/hg19_random.fa",
+    "rundelly_exclude_list": "$HG19_DELLY_ROOT/human.hg19.excl.tsv"
+  },
+   "hg38": {
+    "rundelly_module": "delly/0.9.1 bcftools/1.9 tabix/0.2.6 hg38/p12 hg38-delly/1.0",
+    "rundelly_fasta": "$HG38_ROOT/hg38_random.fa",
+    "rundelly_exclude_list": "$HG38_DELLY_ROOT/human.hg38.excl.tsv"
+   }
 }
 
 Array[File] inputBams= select_all([inputTumor,inputNormal])
-String sampleID = if outputFileNamePrefix=="" then basename(inputBams[0], ".bam") else outputFileNamePrefix
```

Comment:
I understand that we need to supply an output Prefix, but removing this makes it impossible to run delly without specifying a prefix. Not sure if this is how it should be done

Reply:
Whether to make it necessary to supply an outputprefix as input is worth discussion here. My understanding is making the output file names specific to the samples is a pro for doing this.